test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -1811,7 +1811,8 @@ TEST_P(BasicTest, BackendsRestart) {
   CheckRpcSendFailure();
   // Restart all backends.  RPCs should start succeeding again.
   StartAllBackends();
-  CheckRpcSendOk(1, RpcOptions().set_timeout_ms(2000).set_wait_for_ready(true));
```

Comment:
I assume that this RPC was failing because it was hitting its deadline?  If so, let's just increase the deadline here.

If the RPC was failing with something other than DEADLINE_EXCEEDED, then there's a real bug here that we need to find and fix, because that should never happen for a `wait_for_ready` RPC in this scenario.

Reply:
The error is 14: UNAVAILABLE.

Per discussion, restore the use of wait_for_ready; instead of sending multiple RPC to get a success, add more failure RPCs during shutdown before restarting.  Send at least as many RPCs as there are backends.

Ran the test 1000 without failures.